### ISSUE TITLE
Loss computation in train.py

### ISSUE THREAD

**Issue by torki-hossein** [dvl-tum/group_loss#4]
Hi, Thanks for sharing your good code, I have a problem with the loss mentioned in the train.py file, it's here:
# compute the losses
loss1 = criterion(probs_for_gtg, Y)
loss2 = criterion2(probs, Y)
loss = args.scaling_loss * loss1 + loss2

you can also find the definition of criterion(NLLLoss) and criterion2(CELoss) in this file.

1- As you mentioned in the original paper, group loss is not depend on the other loss and can be computed independently, so why you add classic classification CE loss in addition with group loss? or can you refer me (Original paper) to where you define the loss like what implemented in train.py?
2-Is the group loss depended to other loss?
3-Is it any problem for updating parameters using only group loss in the backward process?
Thanks.

**Comment by TheRevanchist** [comment]
Hi,

In general, you get slightly lower results (around 1 point or so) if you do not use the auxiliary loss function. So, while the group loss can be trained without the auxillary loss, it seems that adding it slightly improves the results.

**Comment by torki-hossein** [comment]
> Hi,
> 
> In general, you get slightly lower results (around 1 point or so) if you do not use the auxiliary loss function. So, while the group loss can be trained without the auxillary loss, it seems that adding it slightly improves the results.

Thanks a lot.
